### ISSUE TITLE
XCFramework fails with x86 and arm64 pod spec lint

### ISSUE THREAD

**Issue by nikiizvorski** [CocoaPods/CocoaPods#11065]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I have a pod currently available here: https://github.com/nikiizvorski/TestFramework and this is my podspec file:

```
Pod::Spec.new do |spec|
  spec.name         = 'TestLibN'
  spec.version      = '1.0.5'
  spec.license      = 'MIT'
  spec.summary      = 'Sample library'
  spec.homepage     = 'https://github.com/nikiizvorski/TestFramework'
  spec.author       = 'Niki Izvorski'
  spec.source       = { :git => 'git://github.com/nikiizvorski/TestFramework.git', :tag => '1.0.5' }
  spec.vendored_frameworks = 'TestLib.xcframework'
  spec.ios.deployment_target = '14.5'
  spec.dependency 'SQLite.swift', '~> 0.13.0'
  spec.dependency 'SwiftCBOR'
  spec.dependency 'SwiftProtobuf', '~> 1.0'
end

```
This is the script i am using to build the XCFramework which locally works fine but when i try to pod spec lint it shows an error and then when i install it with pod install in the client app it makes the same problem with the library missing:

# create archives for each platform

```

xcodebuild archive \
-workspace TestLib.xcworkspace \
-scheme TestLib \
-configuration Release \
-destination 'generic/platform=iOS' \
-archivePath './build/TestLib.framework-iphoneos.xcarchive' \
SKIP_INSTALL=NO \
BUILD_LIBRARIES_FOR_DISTRIBUTION=YES

xcodebuild archive \
-workspace TestLib.xcworkspace \
-scheme TestLib \
-configuration Release \
-destination 'generic/platform=iOS Simulator' \
-archivePath './build/TestLib.framework-iphonesimulator.xcarchive' \
SKIP_INSTALL=NO \
BUILD_LIBRARIES_FOR_DISTRIBUTION=YES

xcodebuild archive \
-workspace TestLib.xcworkspace \
-scheme TestLib \
-configuration Release \
-destination 'platform=macOS,arch=x86_64,variant=Mac Catalyst' \
-archivePath './build/TestLib.framework-catalyst.xcarchive' \
SKIP_INSTALL=NO \
BUILD_LIBRARIES_FOR_DISTRIBUTION=YES

# remove old xcframework
rm -rf ./build/TestLib.xcframework

# create xcframework
xcodebuild -create-xcframework \
-framework './build/TestLib.framework-iphonesimulator.xcarchive/Products/Library/Frameworks/TestLib.framework' \
-framework './build/TestLib.framework-iphoneos.xcarchive/Products/Library/Frameworks/TestLib.framework' \
-framework './build/TestLib.framework-catalyst.xcarchive/Products/Library/Frameworks/TestLib.framework' \
-output './build/TestLib.xcframework'

# remove archives
rm -rf ./build/TestLib.framework-catalyst.xcarchive
rm -rf ./build/TestLib.framework-iphoneos.xcarchive
rm -rf ./build/TestLib.framework-iphonesimulator.xcarchive

```

If i add the XCFramework manually it appears to work. But if i do it over pod install it is a problem.

ℹ Please replace these two lines with what you did.  

## What did you expect to happen?

I expect that the XCFramework should pass lint without any issues or show me some way to fix it. 

ℹ Please replace these two lines with what you expected to happen.  
e.g. Install all pod dependencies correctly.

## What happened instead?

Release-iphonesimulator/App.build/Objects-normal/x86_64/App_dependency_info.dat -o /Users/nikiizvorski/Library/Developer/Xcode/DerivedData/App-bwkmncufxemwfvglahdfyfxpftjd/Build/Products/Release-iphonesimulator/App.app/App
    ld: framework not found TestLib.xcframework

and i believe i have the same issues for simulator arm64 and x86_64 the xcframework should also contain macOS versions.

ℹ Please replace these two lines with of what happened instead.  
e.g. Pod A is missing the subspec B for target C.

## CocoaPods Environment

ℹ Please replace these two lines with the output of `pod env`.

### Stack

```
   CocoaPods : 1.8.4
        Ruby : ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin17]
    RubyGems : 2.7.6
        Host : macOS 11.2.1 (20D75)
       Xcode : 12.5.1 (12E507)
         Git : git version 2.15.1
Ruby lib dir : /usr/local/Cellar/ruby/2.5.0_2/lib
Repositories : master - git - https://github.com/CocoaPods/Specs.git @ 5f8842e0bbff0b0f7f34d12bf326dec1e61bd604
               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-packager    : 1.5.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.4.1
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue

ℹ Please link to a project we can download that reproduces the issue.

I have added links and information how to reproduce. I would be really happy if i could solve this. Thank you! And if i could get some help and advice on the subject which will be highly appreciated!

**Comment by dnkoutso** [comment]
Can you please try CocoaPods 1.11.2? The version you are using is very old and I do not think it handles xcframeworks.

**Comment by nikiizvorski** [comment]
Hi there. I have updated to that version and the issue was gone i am looking at any potential issue after that. As soon as i verify the behaviour will update it and we should be able to close it. Thank you!

**Comment by nikiizvorski** [comment]
I am at the last step pretty much and just pushed another update on the pod 1.0.5 waiting to see if i could install it. I got the following report:

After publishing the latest update:

 🎉  Congrats

 🚀  TestLibN (1.0.5) successfully published
 📅  November 22nd, 13:16
 🌎  https://cocoapods.org/pods/TestLibN
 👍  Tell your friends!


pod trunk info TestLibN and i got:

TestLibN
    - Versions:
      - 1.0.2 (2021-11-19 13:30:22 UTC)
      - 1.0.5 (2021-11-22 11:16:22 UTC)
    - Owners:

But when i do pod install i get the following:

Downloading dependencies
Installing SQLite.swift (0.13.1)
Installing SwiftCBOR (0.4.4)
Installing SwiftProtobuf (1.18.0)
Installing TestLibN (1.0.2)
Generating Pods project
Integrating client project
Pod installation complete! 

And the previous tags aren't actually existing on the cocoapods website it just points to the repo of the library. But the last tag was added which i believe is good
